### PR TITLE
Add usePagError hook tests

### PR DESCRIPTION
**Pull request by myty**
Adds test for issue #22 

- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [ ] Documentation updated (readme, docs, comments, etc.)
- [ ] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -3,7 +3,7 @@ root = true
 
 [*]
 indent_style = space
-indent_size = 2
+indent_size = 4
```
Comment (brandongregoryscott) [inline .editorconfig:6]: Good catch! 🤔

File: src/hooks/use-page-errors.test.tsx [hunk 2]
```diff
@@ -1,3 +1,86 @@
...
+            const errorMessage = "Error Message";
+            const { result } = renderHook(() => usePageErrors());
+
+            // Act
+            act(() => {
+                result.current.setPageErrors([errorMessage]);
+            });
+
+            // Assert
+            expect(result.current.pageErrors.length).toBe(1);
```
Comment (brandongregoryscott) [inline src/hooks/use-page-errors.test.tsx:28]: Minor - I believe we should have a `toHaveLength` matcher

File: src/hooks/use-page-errors.test.tsx [hunk 3]
```diff
@@ -1,3 +1,86 @@
...
+                result.current.handlePageLoadError(error);
+            });
+
+            // Assert
+            expect(result.current.pageErrors.length).toBe(1);
+            expect(result.current.pageErrors[0]).toBe(errorMessage);
+        });
+    });
+
+    describe("resetPageErrors", () => {
```
Comment (brandongregoryscott) [inline src/hooks/use-page-errors.test.tsx:69]: Minor - consider wrapping these describe blocks in our `ts-region` snippet

File: src/hooks/use-page-errors.test.tsx [hunk 4]
```diff
@@ -1,3 +1,86 @@
...
+
+        test("when error is ResultRecord, pageErrors returns array", async () => {
+            // Arrange
+            const errorMessage = "Error Message";
+            const { result } = renderHook(() => usePageErrors());
+
+            // Act
+            act(() => {
+                const error = {
+                    listErrorMessages: () => [errorMessage],
```
Comment (brandongregoryscott) [inline src/hooks/use-page-errors.test.tsx:57]: Consider creating an actual instance of this record vs. casting a POJO as it. Ideally, we'll probably want to promote the `ResultRecordFactory` from `andculturecode-javascript-core` to `andculturecode-javascript-testing` for reuse, but that doesn't have to happen as part of this.

File: src/hooks/use-page-errors.test.tsx [hunk 5]
```diff
@@ -1,3 +1,86 @@
...
+            });
+
+            // Assert
+            expect(result.current.pageErrors.length).toBe(1);
+            expect(result.current.pageErrors[0]).toBe(errorMessage);
+        });
+    });
+
+    describe("handlePageLoadError", () => {
+        test("when error is string, pageErrors returns array", async () => {
```
Comment (brandongregoryscott) [inline src/hooks/use-page-errors.test.tsx:34]: Just curious - do these test blocks need to be `async`? I'm not seeing any `await`s in here

File: src/hooks/use-page-errors.ts [hunk 6]
```diff
@@ -7,14 +7,17 @@ import { ResultRecord } from "andculturecode-javascript-core";
...
     const [pageErrors, setPageErrors] = useState<Array<string>>([]);
 
-    const handlePageLoadError = useCallback((result: any) => {
-        if (result instanceof ResultRecord) {
-            setPageErrors((e) => [...e, ...result.listErrorMessages()]);
-            return;
-        }
+    const handlePageLoadError = useCallback(
+        (result: string | ResultRecord<any>) => {
+            if (typeof result === "string") {
```
Comment (brandongregoryscott) [inline src/hooks/use-page-errors.ts:12]: Thanks for adding some extra safeguards to this function 👍

**Review by brandongregoryscott — APPROVED** [review]
Some minor comments for cleanup if you'd like, but looks good overall. Thanks for taking the time to test this hook, and it was cool to see it tested with `renderHook`! Seems to clean up the boilerplate of spinning up a test component

File: src/hooks/use-page-errors.test.tsx (hunk 5 above)
Comment (myty) [reply src/hooks/use-page-errors.test.tsx:34]: Absolutely not.  Good catch! Not sure how that got in there.

**Comment by myty** [comment]
@brandongregoryscott I think I addressed all of your comments. Thanks!

**Review by brandongregoryscott — APPROVED** [review]

**Pull request merged**